src/libutil/json-utils.cc

```diff
@@ -22,22 +25,103 @@ const nlohmann::json & valueAt(
     const std::string & key)
 {
     if (!map.contains(key))
-        throw Error("Expected JSON object to contain key '%s' but it doesn't", key);
+        throw Error("Expected JSON object to contain key '%s' but it doesn't: %s", key, map.dump());
 
     return map[key];
 }
 
+const nlohmann::json & valueAt(
+    const nlohmann::json::object_t & map,
+    const std::string & key)
+{
+    if (!map.contains(key))
+        throw Error("Expected JSON object to contain key '%s' but it doesn't: %s", key, nlohmann::json(map).dump());
+
+    return map.at(key);
+}
+
+std::optional<nlohmann::json> getNullable(const nlohmann::json & value, const std::string & key)
+{
+    try {
+        auto & v = valueAt(value, key);
+        return v.get<nlohmann::json>();
+    } catch (...) {
+        return std::nullopt;
+    }
+}
```

Comment:
Ah this is different thing than the one I indented, which would turn `null` to `std::nullopt` and anything else to `std::optional { <original> }`.

This one is useful too, but maybe lets call it `optionalValueAt` or something. "nullable" implies `null` but this one isn't about `null` at all! Just about potentially-missing fields.

Reply:
Also, I hope we will transfer from having fewer sometimes-there-sometimes-missing fields, and more explicit `null`s, so this function is hopefully just used for things we don't control and some backwards compat.

Reply:
Ah okay -- makes sense! Thanks :D